crates/services/gas_price_service/src/v1/uninitialized_task.rs

```diff
@@ -0,0 +1,361 @@
+
+    sync_v1_metadata(
+        settings,
+        on_chain_db,
+        metadata_height,
+        latest_block_height,
+        &mut algo_updater,
+        metadata_storage,
+    )?;
+
+    Ok(())
+}
+
+fn sync_v1_metadata<L2DataStore, L2DataStoreView, Metadata, SettingsProvider>(
+    settings: &SettingsProvider,
+    on_chain_db: &L2DataStoreView,
+    metadata_height: u32,
+    latest_block_height: u32,
+    updater: &mut AlgorithmUpdaterV1,
+    metadata_storage: &mut Metadata,
+) -> anyhow::Result<()>
+where
+    L2DataStore: L2Data,
+    L2DataStoreView: AtomicView<LatestView = L2DataStore>,
+    Metadata: MetadataStorage,
+    SettingsProvider: GasPriceSettingsProvider,
+{
+    let first = metadata_height.saturating_add(1);
+    let view = on_chain_db.latest_view()?;
+    for height in first..=latest_block_height {
+        let block = view
+            .get_block(&height.into())?
+            .ok_or(not_found!("FullBlock"))?;
+        let param_version = block.header().consensus_parameters_version;
+
+        let GasPriceSettings {
+            gas_price_factor,
+            block_gas_limit,
+        } = settings.settings(&param_version)?;
+        let block_gas_capacity = block_gas_limit.try_into()?;
+
+        let block_gas_used =
+            match get_block_info(&block, gas_price_factor, block_gas_limit)? {
+                BlockInfo::GenesisBlock => {
+                    Err(anyhow::anyhow!("should not be genesis block"))?
+                }
+                BlockInfo::Block { gas_used, .. } => gas_used,
+            };
+
+        let block_bytes = block_bytes(&block);
+        let (fee_wei, _) = mint_values(&block)?;
+        updater.update_l2_block_data(
+            height,
+            block_gas_used,
+            block_gas_capacity,
+            block_bytes,
+            fee_wei.into(),
+        )?;
+        let metadata: UpdaterMetadata = updater.clone().into();
+        metadata_storage.set_metadata(&metadata)?;
+    }
+
+    Ok(())
+}
+
+#[allow(clippy::type_complexity)]
+#[allow(clippy::too_many_arguments)]
+pub fn new_gas_price_service_v1<
+    L2DataStore,
+    L2DataStoreView,
+    GasPriceStore,
+    Metadata,
+    DA,
+    SettingsProvider,
+>(
+    config: GasPriceServiceConfig,
+    genesis_block_height: BlockHeight,
+    settings: SettingsProvider,
+    block_stream: BoxStream<SharedImportResult>,
+    gas_price_db: GasPriceStore,
+    metadata: Metadata,
+    da_source: DA,
+    on_chain_db: L2DataStoreView,
+) -> anyhow::Result<
+    ServiceRunner<
+        UninitializedTask<L2DataStoreView, GasPriceStore, Metadata, DA, SettingsProvider>,
+    >,
+>
+where
+    L2DataStore: L2Data,
+    L2DataStoreView: AtomicView<LatestView = L2DataStore>,
+    GasPriceStore: GasPriceData,
+    SettingsProvider: GasPriceSettingsProvider,
+    Metadata: MetadataStorage,
+    DA: DaBlockCostsSource,
+{
+    let v1_config = config.v1().ok_or(anyhow::anyhow!("Expected V1 config"))?;
```

Comment:
Are you saying that we should pass `GasPriceV0Config` in some cases?

Reply:
I'm saying the `new_gas_price_service_v1` should expect `V1AlgorithmConfig` to avoid `config.v1().ok_or(anyhow::anyhow!("Expected V1 config"))?` conversion.

The `fuel-core` should decided which gas price to use(0 or 1) based on the enum variant for the `GasPriceServiceConfig`.